### ISSUE TITLE
Add possibility to exclude files from parser but not from file list in Workspace.

### ISSUE THREAD

**Issue by arathburgos05** [microsoft/vscode-cpptools#11609]
### Feature Request

# 📄 Brief introduction
I'm working in c code-generation for that reason I have files in format *.c with invalid proprietary language to enable/disable specific section of code from another configuration file. Also I have the testing environment in the same workspace.

I would like to know if there is any way to exclude specific files/folders using glob patterns **but not hiding them from workspace** (only to the parse ignore or hide problems in file). I tried with:
```
"files.readonlyInclude": {
  "*/oneFolder/anotherFolder/**": true
},
"C_Cpp.codeAnalysis.exclude": {
  "*/oneFolder/anotherFolder/**": true
},
"C_Cpp.files.exclude": {
  "*/oneFolder/anotherFolder/**": true
},
"search.exclude": {
  "*/oneFolder/anotherFolder/**": true
},
"files.exclude": {
  "*/oneFolder/anotherFolder/**": true
}
```

Last one could work but it will hide my files from workspace and I want to keep them.

# ⚙ Use case
To consult files and not having errors in problems section when the file is expected to have a wrong format (in this case a code-generated c source). For example:
![image](https://github.com/microsoft/vscode/assets/25379464/b892023a-d79b-423e-b8b1-7b85e54cfa5e)

# 🛠 Possible solution
A setting similar to `C_Cpp.files.exclude` or to `files.exclude` which won't report any kind of "problem" related to parser or wrong syntax.

**Comment by sean-mcmanus** [comment]
Use folder-based exclusion instead of file-based exclusion.
```json
"C_Cpp.files.exclude": {
  "*/oneFolder/anotherFolder": true
},
```

Does that work for you?

Using the `**` at the end would require C_Cpp.exclusionPolicy to be changed (not recommended).

**Comment by sean-mcmanus** [comment]
@arathburgos05 Okay, looks like that only works for tag parsing if the file isn't open, but it's still parsed by IntelliSense if the file is open, i.e. we'd have to start to check that exclusion for IntelliSense or add a C_Cpp.intelliSense.exclude setting.

**Comment by sean-mcmanus** [comment]
@arathburgos05 Yeah, I also wanted to reference a specific workspace folder in the various excludes settings -- it's being tracked by VS Code issue https://github.com/microsoft/vscode/issues/50638

**Comment by arathburgos05** [comment]
> @arathburgos05 Okay, looks like that only works for tag parsing if the file isn't open, but it's still parsed by IntelliSense if the file is open, i.e. we'd have to start to check that exclusion for IntelliSense or add a C_Cpp.intelliSense.exclude setting.
> 
> You could potentially exclude those particular folders with files.exclude and then re-add them back as a workspace folder with C_Cpp.intelliSenseEngine set to "Tag Parser" for those folders (you'll still get missing include blue/info warnings though).

thanks @sean-mcmanus , is there any open PR where is working " C_Cpp.intelliSense.exclude setting" feature? I want to exclude files from intellisense indexer but not from my workspace :(

Regards.

**Comment by sean-mcmanus** [comment]
@arathburgos05 There's no open PR currently, but it should be a trivial issue for us to fix, possibly even just re-using the C_Cpp.files.exclude setting, which is inherited from other settings like C_Cpp.codeAnalysis.exclude.

**Comment by sean-mcmanus** [comment]
@Colengms I currently making changes in this area (IntelliSense updating), so maybe he could add this as part of that larger change for 1.19.0.
